packages/@aws-cdk/aws-ec2/test/integ.vpc-lookup.ts

```diff
@@ -2,24 +2,31 @@ import * as cdk from '@aws-cdk/core';
 import { IntegTest } from '@aws-cdk/integ-tests';
 import * as ec2 from '../lib';
 
-const appWithVpc = new cdk.App();
-const stack = new cdk.Stack(appWithVpc, 'StackWithVpc', {
+const app = new cdk.App();
+const account = process.env.CDK_INTEG_ACCOUNT ?? process.env.CDK_DEFAULT_ACCOUNT;
+/**
+ * Since the VPC lookup happens at synthesis time, a manual setup is required beforehand to run this.
+ *
+ * The following stack with its VPC has to be deployed, so that the context can be filled. Otherwise the context
+ * must be set manually so that the lookup works.
```

Comment:
I don't follow, why doesn't adding a dependency from the stack doing the lookup to the stack that defines the VPC enough?

Reply:
We want to avoid required manual setup to deploy these wherever possible

Reply:
@comcalvi I tried adding the other stack as dependency (with `addDependency`), but I think the lookup is done before the deployment (at synthesis time).

If you know an option how I can deploy the stack before I would be really thankful! I would prefer no manual steps either.